nautobot/dcim/views.py

```diff
@@ -734,81 +760,669 @@ class DeviceTypeUIViewSet(NautobotUIViewSet):
+                                label="Delete",
+                                color=ButtonActionColorChoices.DELETE,
+                                icon="mdi-trash-can-outline",
+                                size="xs",
+                                form_id="interfacetemplate_form",
+                                weight=400,
+                            ),
+                        ],
+                    ),
+                ),
+            ),
+            object_detail.DistinctViewTab(
+                weight=200,
+                tab_id="frontports",
+                label="Front Ports",
+                url_name="dcim:devicetype_frontports",
+                related_object_attribute="front_port_templates",
+                panels=(
+                    object_detail.ObjectsTablePanel(
+                        section=SectionChoices.FULL_WIDTH,
+                        weight=100,
+                        table_title="Front Ports",
+                        table_class=tables.FrontPortTemplateTable,
+                        table_filter="device_type",
+                        tab_id="frontports",
+                        enable_bulk_actions=True,
+                        form_id="frontporttemplate_form",
+                        footer_buttons=[
+                            object_detail.FormButton(
+                                link_name="dcim:frontporttemplate_bulk_rename",
+                                link_includes_pk=False,
+                                label="Rename",
+                                color=ButtonActionColorChoices.RENAME,
+                                icon="mdi-pencil",
+                                size="xs",
+                                form_id="frontporttemplate_form",
+                                weight=200,
+                            ),
+                            object_detail.FormButton(
+                                link_name="dcim:frontporttemplate_bulk_edit",
+                                link_includes_pk=False,
+                                label="Edit",
+                                color=ButtonActionColorChoices.EDIT,
+                                icon="mdi-pencil",
+                                size="xs",
+                                form_id="frontporttemplate_form",
+                                weight=300,
+                            ),
+                            object_detail.FormButton(
+                                link_name="dcim:frontporttemplate_bulk_delete",
+                                link_includes_pk=False,
+                                label="Delete",
+                                color=ButtonActionColorChoices.DELETE,
+                                icon="mdi-trash-can-outline",
+                                size="xs",
+                                form_id="frontporttemplate_form",
+                                weight=400,
+                            ),
```

Comment:
Can we create some reusable base classes (perhaps `BulkRenameButton`, `BulkEditButton`, `BulkDeleteButton`) and use them to reduce the amount of repeated code in this object_detail definition? Ideally I'd like to see something more like:

```py
footer_buttons=[
    BulkRenameButton(form_id="frontporttemplate_form", model=FrontPortTemplate),
    BulkEditButton(form_id="frontporttemplate_form", model=FrontPortTemplate),
    BulkDeleteButton(form_id="frontporttemplate_form", model=FrontPortTemplate),
]
```